RealmSwift/Object.swift

```diff
@@ -0,0 +1,296 @@
+        if let list = listProperty(key) {
+            return list
+        }
+        return super.valueForKey(key)
+    }
+
+    public override func setValue(value: AnyObject?, forKey key: String) {
+        if let list = listProperty(key) {
+            if let value = value as? NSFastEnumeration {
+                list._rlmArray.removeAllObjects()
+                list._rlmArray.addObjects(value)
+            }
+            return
+        }
+        super.setValue(value, forKey: key)
+    }
+
+    public subscript(key: String) -> AnyObject? {
+        get {
+            if let list = listProperty(key) {
+                return list
+            }
+            return RLMObjectBaseObjectForKeyedSubscript(self, key)
+        }
+        set(value) {
+            if let list = listProperty(key) {
+                if let value = value as? NSFastEnumeration {
+                    list._rlmArray.removeAllObjects()
+                    list._rlmArray.addObjects(value)
+                }
+                return
+            }
+            RLMObjectBaseSetObjectForKeyedSubscript(self, key, value)
+        }
+    }
+
+    // Helper for getting a list property for the given key
+    private func listProperty(key: String) -> RLMListBase? {
+        if let prop = RLMObjectBaseObjectSchema(self)?[key] {
+            if prop.type == .Array {
+                return object_getIvar(self, prop.swiftListIvar) as RLMListBase?
+            }
+        }
+        return nil
+    }
+}
+
+// MARK: Equatable
+
+/// Returns whether both objects are equal.
+public func == <T: Object>(lhs: T, rhs: T) -> Bool {
+    return RLMObjectBaseAreEqual(lhs, rhs)
+}
+
+/// Object interface which allows untyped getters and setters for Objects.
+public final class DynamicObject : Object {
+    private var listProperties = [String: List<DynamicObject>]()
+
+    /// Returns or sets the value of the property with the given name.
+    public override subscript(key: String) -> AnyObject? {
+        get {
+            if let prop = RLMObjectBaseObjectSchema(self)[key] {
+                if prop.type == .Array {
+                    // get it or set it
+                    if let list = listProperties[key] {
+                        return list
+                    }
+                    let list = List<DynamicObject>()
+                    listProperties[key] = list
+                    return list
+
+                }
+            }
+            return RLMObjectBaseObjectForKeyedSubscript(self, key)
+        }
+        set(value) {
+            if let prop = RLMObjectBaseObjectSchema(self)[key] {
+                if prop.type == .Array {
+                    throwRealmException("Setting List properties is unsupported. Instead you can add or remove objects from the current List.")
+                }
+            }
+            RLMObjectBaseSetObjectForKeyedSubscript(self, key, value)
+        }
+    }
+}
+
+
+// Internal class. Do not use directly.
```

Comment:
As it stands now, since it's a public declaration, it'll show up in the jazzy-generated docs with no indication whatsoever that it's a class for internal use only. There's an open issue on jazzy to explicitly ignore certain tokens, but that's not done yet: https://github.com/realm/jazzy/issues/67.

Even jazzy aside, this class will show up in Xcode's autocomplete and without a documentation comment, there won't be any indication that it's a private class.